packages/data-stores/src/verticals-queries/use-vertical-images-query.ts

```diff
@@ -0,0 +1,23 @@
+import { stringify } from 'qs';
+import { useQuery, UseQueryResult, QueryKey } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { VerticalImage } from './types';
+
+export function useVerticalImagesQuery( id: string ): UseQueryResult< VerticalImage[] > {
+	return useQuery( getCacheKey( id ), () => fetchVerticalImages( id ), {
+		enabled: typeof id === 'string' && id !== '',
+		staleTime: Infinity,
+	} );
+}
+
+function fetchVerticalImages( id: string ): Promise< VerticalImage[] > {
+	return wpcomRequest< VerticalImage[] >( {
+		apiNamespace: 'wpcom/v2',
+		path: `/site-verticals/${ id }/images`,
+		query: stringify( { include_parents: true } ),
```

Comment:
Could we also limit the number of images (Still need to implement)?. I'm afraid there are too many images and we don't need them actually. It might be better to only fetch the count but I think we'll create a new endpoint like `GET /generated-designs` to handle this kind of logic. So, it's okay to fetch the images for now.

Reply:
Do you suggest:
1) Adding a new endpoint `/site/verticals/${id}/images/count`, or
2) Adding a new parameter to existing endpoint to limit number of images

Reply:
I'm okay with both 🙂

Reply:
Hey @taipeicoder @arthur791004 if we're going to need to modify something on the backend side anyway (which is potentially unused post-MVP), what if we create that new `/generated-designs` right now? That would be MUCH cleaner 😄 right now the changes make the logic in Design Picker a bit complicated (it already is right now 😅)

Reply:
+1 but we need to find bandwidth for it, I think for now the three of us have higher priority things to tackle one. I might be able to do it next week.

Reply:
Let me try to write up a ticket for that! (Nowadays I kind of love doing that...)

Reply:
Added this issue: https://github.com/Automattic/wp-calypso/issues/63485